extensions/iceberg/src/main/java/io/deephaven/iceberg/location/IcebergTableParquetLocation.java

```diff
@@ -0,0 +1,96 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.location;
+
+import io.deephaven.api.ColumnName;
+import io.deephaven.api.SortColumn;
+import io.deephaven.engine.table.impl.locations.TableKey;
+import io.deephaven.engine.table.impl.locations.TableLocation;
+import io.deephaven.iceberg.util.IcebergTableAdapter;
+import io.deephaven.parquet.table.ParquetInstructions;
+import io.deephaven.parquet.table.location.ParquetTableLocation;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.NullOrder;
+import org.apache.iceberg.Schema;
+import org.apache.iceberg.SortDirection;
+import org.apache.iceberg.SortField;
+import org.apache.iceberg.SortOrder;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class IcebergTableParquetLocation extends ParquetTableLocation implements TableLocation {
+
+    private volatile List<SortColumn> sortedColumns;
+
+    public IcebergTableParquetLocation(
+            @NotNull final IcebergTableAdapter tableAdapter,
+            @NotNull final TableKey tableKey,
+            @NotNull final IcebergTableParquetLocationKey tableLocationKey,
+            @NotNull final ParquetInstructions readInstructions) {
+        super(tableKey, tableLocationKey, readInstructions);
+        sortedColumns = computeSortedColumns(tableAdapter, tableLocationKey.dataFile());
+    }
+
+    @Override
+    @NotNull
+    public List<SortColumn> getSortedColumns() {
+        List<SortColumn> local;
+        if ((local = sortedColumns) != null) {
+            return local;
+        }
+        synchronized (this) {
+            if ((local = sortedColumns) != null) {
+                return local;
+            }
+            local = Collections.unmodifiableList(super.getSortedColumns());
+            sortedColumns = local;
+            return local;
+        }
```

Comment:
I don't see much value in having this be volatile and caching `super.getSortedColumns()`.

It should be just as good to do

```java
public List<SortColumn> getSortedColumns() {
  return sortedColumns == null ? super.getSortedColumns() : sortedColumns;
}
```
, yeah?